mod/state-transition/pkg/core/validation_deposits.go

```
@@ -43,8 +43,16 @@ func (sp *StateProcessor[
 
 	case sp.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID:
 		// Boonet inherited the bug from Bartio and it may have added some
-		// validators before we activate the fork. So we skip validation
-		// before fork activation
+		// validators before we activate the fork. So we skip all validations
+		// but the validator set cap.
+		//#nosec:G701 // can't overflow.
+		if uint32(len(deposits)) > sp.cs.GetValidatorSetCap() {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int -> uint32

(gosec)